components/camera/videosource/logging/logger.go

```diff
@@ -0,0 +1,248 @@
+// Package logger is a thread-safe way to log video device information to a file. On startup, this package creates a
+// unique filename and uses that filename throughout the lifetime of the program to log information such as which video
+// devices are V4L2 compatible and the current operating system.
+//
+// To use this logger call Logger.Add to add a dependency, Logger.Log to log additional information, and Logger.Done
+// when you will no longer need to Logger.Log anything. Most likely you'll want to call Logger.Add during the Open method
+// and Logger.Done on the Close method of the dependent object. All public methods are thread-safe, so they may block
+// while waiting their turn to log information.
+package logger
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/natefinch/lumberjack"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/config"
+)
+
+var filePath string
+
+func init() {
+	t := time.Now().UTC().Format(time.RFC3339)
+	t = strings.ReplaceAll(strings.ReplaceAll(t, ":", ""), "-", "")
+	filePath = filepath.Join(config.ViamDotDir, fmt.Sprintf("debug_camera_component_%s.txt", t))
+}
+
+// InfoMap is a map of information to be written to the log.
+type InfoMap = map[string]string
+
+// Logger is a thread-safe logger that manages a single log file in config.ViamDotDir.
+type Logger struct {
+	ctx       context.Context
+	nActive   atomic.Int32
+	cancel    func()
+	doneCh    chan struct{}
+	infoCh    chan info
+	Logger    golog.Logger
+	prevInfo  map[string]InfoMap
+	isRunning atomic.Bool
+}
+
+type info struct {
+	title string
+	m     InfoMap
+}
+
+// NewLogger creates a new Logger. No background routines are started until the first entity calls Logger.Add.
+func NewLogger() Logger {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// keeps a rolling log with a size of at most MaxSize
+	lumlog := &lumberjack.Logger{
+		Filename:   filePath,
+		MaxSize:    1, // megabytes
+		MaxBackups: 0, // number of log files
```

Comment:
Do we want to also add `MaxAge` here? If new log files are being generated each RDK run we want stale files to be pruned?